### PR TITLE
Refactor `h.auth` for clarity

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR does some minor cleanup in the `h.auth` module and attempts to add some more full-fledged comments so that future devs (or even me in the future) can have more a clue about how it works.

### CODE REVIEW

File: h/auth/__init__.py [hunk 1]
```diff
@@ -22,24 +41,33 @@
...
+PROXY_FALLBACK_POLICY = RemoteUserAuthenticationPolicy(environ_key='HTTP_X_FORWARDED_USER',
+                                                       callback=groupfinder)
+PROXY_POLICY = AuthenticationPolicy(api_policy=API_POLICY,
+                                    fallback_policy=PROXY_FALLBACK_POLICY)
+
 WEBSOCKET_POLICY = TOKEN_POLICY
 
 
 def includeme(config):
-    global DEFAULT_POLICY
```
Comment (lyzadanger) [inline h/auth/__init__.py]: AFAICT, this is redundant and unused...no grep hits for this anywhere else in the app

File: h/auth/__init__.py [hunk 2]
```diff
@@ -52,19 +80,18 @@ def includeme(config):
...
     if config.registry.settings.get('h.proxy_auth'):
+        # Set a proxy authentication policy instead of the default policy
         log.warning('Enabling proxy authentication mode: you MUST ensure that '
                     'the X-Forwarded-User request header can ONLY be set by '
                     'trusted downstream reverse proxies! Failure to heed this '
                     'warning will result in ALL DATA stored by this service '
                     'being available to ANYONE!')
+        config.set_authentication_policy(PROXY_POLICY)
 
-        DEFAULT_POLICY = AuthenticationPolicy(api_policy=API_POLICY,
```
Comment (lyzadanger) [inline h/auth/__init__.py]: Seemed like we shouldn't be using constant-style var naming if we're reassigning a value here...I've moved all of the "constants" out of the `includeme` function.

**Comment by robertknight** [comment]
It isn't immediately obvious to me what the problem is, but WebSocket authentication doesn't appear to be working for me on this branch whereas it is on master. I'm testing as follows:

1. With h and client running, open http://localhost:3000 in two tabs
2. Switch to a private group in both
3. Create or update annotations in one tab, check that the other tab reports something changed (red indicator arrow at top)

**Comment by lyzadanger** [comment]
@robertknight This PR certainly isn't important enough in our current sprint to warrant anyone tracking that down. I did it when I had some time without internet, but it's not high-priority. I'll close for now.

**Pull request closed without merging**